trieste/acquisition/rule.py

```diff
@@ -444,17 +409,59 @@ def acquire(
 
             is_global = step_is_success or not state.is_global
 
-        if is_global:
-            acquisition_space = search_space
-        else:
-            xmin = dataset.query_points[tf.argmin(dataset.observations)[0], :]
-            acquisition_space = Box(
-                tf.reduce_max([global_lower, xmin - eps], axis=0),
-                tf.reduce_min([global_upper, xmin + eps], axis=0),
-            )
+            if is_global:
+                acquisition_space = self._global_search_space
+            else:
+                xmin = dataset.query_points[tf.argmin(dataset.observations)[0], :]
+                acquisition_space = Box(
+                    tf.reduce_max([global_lower, xmin - eps], axis=0),
+                    tf.reduce_min([global_upper, xmin + eps], axis=0),
+                )
 
-        acquisition_function = self._builder.prepare_acquisition_function(datasets, models)
-        point = self._optimizer(acquisition_space, acquisition_function)
-        state_ = TrustRegion.State(acquisition_space, eps, y_min, is_global)
+            new_state = ContinuousTrustRegionState(acquisition_space, eps, y_min, is_global)
+            return new_state, acquisition_space
+
+        return go
+
+
+def continuous_trust_region(
+    beta: float = 0.7, kappa: float = 1e-4
+) -> TrustRegion[Box, ContinuousTrustRegionState]:
+    """
+    Implements the *trust region* algorithm for constructing a local acquisition space from a global
+    continuous search space. The algorithm is as follows:
+
+    The local acquisition space is the space constructed on each step by this `TrustRegion`,
+    and can be either the global search space, or a trust region which is typically smaller than
+    the global search space and changes on each step.
 
-        return point, state_
+    If no state is specified (it is `None`), the :attr:`default_state` is used.
```

Comment:
Does this line belong here? By the time we call the continuous trust region we've already defaulted the state, no?

Reply:
yes, good catch